### PR TITLE
Pytest-ify -- AssemblySet and DifferentialExpr... etc.

### PR DESCRIPTION
**Pull request by ialarmedalien**
Updating AssemblySet and DiffExprMatrixSet tests to use pytest

### CODE REVIEW

File: test/AssemblySet_basic_test.py [hunk 1]
```diff
@@ -1,82 +1,82 @@
...
 
-    :param cls: class object
-    :type cls: BaseTestClass
-    """
-    # copy test file to scratch area
-    fna_filename = "seq.fna"
-    fna_path = os.path.join(cls.config["scratch"], fna_filename)
-    shutil.copy(os.path.join(TEST_BASE_DIR, "data", fna_filename), fna_path)
+@pytest.fixture()
+def assembly_refs(
```
Comment (ialarmedalien) [inline test/AssemblySet_basic_test.py:13]: create a fixture for building the assembly refs

File: test/AssemblySet_basic_test.py [hunk 2]
```diff
@@ -1,82 +1,82 @@
...
+
 
-def test_basic_save_and_get(self):
-    setObjName = "set_of_assemblies"
+def test_basic_save_and_get(
+    set_api_client: SetAPI,
+    ctx: dict[str, str | list],
+    ws_name: str,
+    assembly_refs: list[str],
+) -> None:
```
Comment (ialarmedalien) [inline test/AssemblySet_basic_test.py:41]: Add the fixtures required for each test

File: test/AssemblySet_basic_test.py [hunk 3]
```diff
@@ -101,64 +101,66 @@ def test_basic_save_and_get(self):
...
-    setObjName = "nada_set"
+
+def check_empty_set(obj: dict, description: str) -> None:
+    """Check that a set object has zero items in it.
+
+    :param obj: set data as returned by the SetAPI
+    :type obj: dict
+    :param description: description parameter of the set
+    :type description: str
+    """
```
Comment (ialarmedalien) [inline test/AssemblySet_basic_test.py:124]: For this module, I extracted out the zero items set test into its own function. I'm sure there is other refactoring that could be done throughout this test suite but YOLO!

File: test/run_tests.sh [hunk 4]
```diff
@@ -20,4 +20,4 @@ pytest \
     --cov-report=html \
     --cov-report=xml \
     -vv \
-    test/
+    test/AssemblySet_basic_test.py test/DifferentialExpressionMatrixSet_basic_test.py
```
Comment (briehl) [inline test/run_tests.sh:23]: Just adding in the ones that are fixed up for now? Watching that test coverage grow after each PR will be nice. :)

**Review by briehl — APPROVED** [review]
👍

**Pull request merged**